### PR TITLE
PP-9545 Add get agreement from ledger

### PR DESCRIPTION
**Pull request by sfount**
* updates current agreement model to clarify that it is just the
  `ConnectorCreateAgreementResponse` serialiser that gets us the
  agreement ID
* adds Ledger methods to get an agreement
* fixtures and builders for future pacts and current IT

### CODE REVIEW

File: src/test/java/uk/gov/pay/api/it/CreateAgreementIT.java [hunk 1]
```diff
@@ -39,8 +39,7 @@ public void shouldCreateAgreement() {
         postAgreementRequest(agreementPayload(createAgreementRequestParams))
                 .statusCode(HttpStatus.SC_CREATED)
                 .contentType(JSON)
-                .body("agreement_id", is(VALID_AGREEMENT_ID))
-                .body("reference", is(REFERENCE));
+                .body("agreement_id", is(VALID_AGREEMENT_ID));
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/api/it/CreateAgreementIT.java:42]: Why was reference removed?

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks good. Couple of small things noted in the comments and do we need an `GetAgreementExceptionMapper`?

File: src/test/java/uk/gov/pay/api/it/CreateAgreementIT.java (hunk 1 above)
Comment (sfount) [reply src/test/java/uk/gov/pay/api/it/CreateAgreementIT.java:42]: Yeah good question. The idea will be that Connector returns just the resource ID of the thing it has created, the API can then look this up from the event store and respond.

That idea took a bit of time to develop and for some reason the agreement id and reference were parsed from Connector.

**Review by stephencdaly — DISMISSED** [review]
Previous review comments have been addressed

**Pull request merged**